src/components/Topics.jsx

```diff
@@ -1,7 +1,73 @@
-import { Link } from "react-router-dom"
+import { useEffect, useState } from "react";
+import { useSearchParams } from "react-router-dom";
+import { getArticles } from "../../api";
+import AllArticlesCard from "./AllArticlesCard";
 
 export function Topics() {
+
+    const [isLoading, setIsLoading] = useState(false)
+    const [searchParams, setSearchParams] = useSearchParams();
+    const [articles, setArticles] = useState([])
+
+    function handleTopic (event) {
+        setIsLoading(true)
+        const topic = event.target.value
+        setSearchParams({topic: topic})
+    }
+
+    useEffect(() => {
+        setIsLoading(true)
+        getArticles(searchParams.get("topic"))
+        .then((data) => {
+            setArticles(data)
+            setIsLoading(false)
+        })
+        .catch((err) => {
+            console.log(err)
+            setIsLoading(false)
+        })
+    }, [searchParams]) 
+
+
+    if (isLoading) {
+        return (
+            <div className="loadingWheel">
+        <div role="status">
+        <svg aria-hidden="true" className="inline w-60 h-60 text-gray-200 animate-spin dark:text-gray-600 fill-red-600" viewBox="0 0 100 101" fill="none" xmlns="http://www.w3.org/2000/svg">
+            <path d="M100 50.5908C100 78.2051 77.6142 100.591 50 100.591C22.3858 100.591 0 78.2051 0 50.5908C0 22.9766 22.3858 0.59082 50 0.59082C77.6142 0.59082 100 22.9766 100 50.5908ZM9.08144 50.5908C9.08144 73.1895 27.4013 91.5094 50 91.5094C72.5987 91.5094 90.9186 73.1895 90.9186 50.5908C90.9186 27.9921 72.5987 9.67226 50 9.67226C27.4013 9.67226 9.08144 27.9921 9.08144 50.5908Z" fill="currentColor"/>
```

Comment:
This is a lot to look at! I'd probably save this information to another file and bring it in. It makes the code hard to read